### DISCUSSION TITLE
Thunder General Discussion

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by realeatham** [comment]
You should make a Lemmy community for this. Also check issues the android app can't access the internet

**Reply by hjiangsu** [reply]
Here's the lemmy community! I'll add this to the README as well for better discovery.

https://lemmy.world/c/thunder_app

**Comment by guigs4** [comment]
Hi,
Are you working on comment voting?
If not, I might create a PR soon to address this.

**Reply by hjiangsu** [reply]
Hey, yeah, I'm currently working on some stuff related to comment voting and saving! It'll be similar to the Apollo swipe gestures (if you know how that looks like)

**Reply by guigs4** [reply]
Yep, I'm a former Apollo user.

Here's how my implementation is going:

![image](https://github.com/hjiangsu/thunder/assets/53377890/21be8160-9a9d-4a1e-9ce2-2d745620dcbb)

Currently it does properly upvote/downvote and update, as well as highlight which (if either) you've chosen.

![image](https://github.com/hjiangsu/thunder/assets/53377890/657225d9-84c9-4be5-8640-9a79b4910b19)

Still needs some improvements though:
The gesture is bit inconvenient (you need to slide, then tap as opposed to Apollo's slide).
As the size of each ActionPane is relative to screen width, they look stretched on wide views.
The code behind needs some cleanup.
My color choices are bad and temporary. 
And so on.

If you want I could submit the PR just to get the feature out there until you finish your implementation.


Bonus: I also added a reply button (which currently does nothing :D)
![image](https://github.com/hjiangsu/thunder/assets/53377890/9277e06b-befd-404b-b1fd-7982fbf993c3)

**Reply by realeatham** [reply]
When it's added there should be an option to not have swipe gestures and use buttons tho

**Reply by guigs4** [reply]
I mean, Apollo does support just taping the score to cycle between votes (no vote -> upvote -> downvote) and was something I was intending to tackle after refactoring.

Good suggestion nonetheless.

**Reply by hjiangsu** [reply]
Hey, thanks for helping contribute to the project! That looks pretty close to what i have so far! This is a preview of my implementation (i'll have the implementation pushed to the `develop` branch here if you are curious

Initially, I implemented it similar to yours, where you would need to slide and tap on the action. After doing a bit of a deeper implementation, I managed to get rid of that need of tapping, so it just requires you to slide to the appropriate "selection"

I have not yet implemented anything with regards to comment voting outside of the slide gestures (e.g., tapping on the score) if that's something that youre interested in implementing!


![comment_voting](https://github.com/hjiangsu/thunder/assets/30667958/0444713d-2c48-4dde-ab97-604b44f01884)

**Reply by jordan1776** [reply]
really slick, very minor thing, the colors on Lemmy for the votes are reversed from reddit (blue #007bff for up and dark orange #873208 from inspect element)

**Reply by hjiangsu** [reply]
hmm yeah, I've noticed that - I was wondering whether to keep it as the reddit's colours, or to switch over to lemmy's way. I was thinking initially to just keep it as the reddit colour so that it makes it less confusing for users who are transitioning from the reddit apps

however, i'm open to suggestions on what should be done!

**Reply by jordan1776** [reply]
The colors probably should be a setting in the theming with options for Lemmy, Reddit, and Kbin (green/red), To me, since it is a Lemmy app it should default to match Lemmy's colors. Maybe a OOBE, where when guiding through a user who first installs the app, ask "Are you new to Lemmy?" and if yes, use the Reddit colors?

**Reply by realeatham** [reply]
I think Instead of asking if they are new to Lemmy it could just ask what colors they want, and there should be a custom colors option (and a background color setting)

**Reply by hjiangsu** [reply]
> The colors probably should be a setting in the theming with options for Lemmy, Reddit, and Kbin (green/red), To me, since it is a Lemmy app it should default to match Lemmy's colors. 

Good idea, I think this is easy to implement and allows for some degree of customization - feel free to add this as an issue so that it can be tracked in the future!

**Comment by guigs4** [comment]
Aaaaaand it is done:

![ezgif-2-c8369e6552](https://github.com/hjiangsu/thunder/assets/53377890/35cec217-4eab-45ef-abc6-c4b053a74f4d)

**Reply by hjiangsu** [reply]
awesome, that looks great!

**Comment by guigs4** [comment]
Also: partly solved issue #20, currently I only added the option to hide thumbnails
![image](https://github.com/hjiangsu/thunder/assets/53377890/3d09cd27-1876-4248-8a2f-5314bcc2c896)

No previews of it working for obvious reasons

**Reply by hjiangsu** [reply]
thanks! is it okay if i assign you to that ticket for now? since its partially on the way?

**Reply by guigs4** [reply]
Sure thing!

**Reply by guigs4** [reply]
I will switch the behavior to match Apollo's and then start a PR.

Starting tomorrow I won't be able to reliably work on refining the feature for about a week or so.
Getting the PR out will allow for the feature to get out there for users and for developers to further refine in case I can't get to it in a timely manner.

**Reply by hjiangsu** [reply]
Thats alright! Any help is better than no help 😄 Are you able to leave a comment on #20 so that I can assign the ticket to you for the meantime?

**Comment by shortwavesurfer2009** [comment]
I think Thunder should load more posts at once. I generally like "subscribed" and "new" for my defaults but this causes quite a bit of scrolling and loading after a while away. Sleep for 6-8 hours and then scroll to the posts you have already read. It takes a bit. I think maybe loading 50 at once instead of like 15.

**Comment by guigs4** [comment]
About #126:
Already have it working but it involved manually altering the lemmy_api_client library, It's a small change, just had to add a couple of entries to SortType (enums.dart).
Problem is: this obviously means it'll break should anyone try to use it.
I thought about opening an issue on liftoff-app/lemmy_api_client but they don't seem to have an 'issues' page.
I'll try to open a PR to address this over there.

Just letting y'all know.

**Comment by Tealk** [comment]
Wouldn't it be better if the feature requests were posted here, then they could be voted up and down.

**Reply by hjiangsu** [reply]
I think for feature requests, its okay to have them as issues on GitHub, but then have people vote on them by giving thumbs up for features that they feel like are more important

This way, I (and others in the community) can easily sort by those issues and place a greater focus on them

Thoughts on this?

**Reply by Tealk** [reply]
Ah ok I have not noticed until now that you can also sort issues by votes, then I have said nothing 😅

But then it might make sense to remove [this category](https://github.com/hjiangsu/thunder/discussions/categories/ideas), otherwise the ideas/features end up splitting into issues and discussions.

**Comment by Tealk** [comment]
Another idea would be a chat like Matrix, then you could talk a little bit about it before the bug report and maybe improve the reports?

**Reply by hjiangsu** [reply]
I think it'll be good to have a Matrix chat for general things too, so that it's easier to communicate with others around topics here

I have never used Matrix so this will be something I have to look into!

**Reply by Tealk** [reply]
Well I like to use Matrix and I have my own server, if you have any questions/need help just let me know.

**Reply by hjiangsu** [reply]
I have created a Matrix community for Thunder! https://matrix.to/#/%23thunderapp%3Amatrix.org

There's not much to it but if you have any suggestions, feel free to let me know! If there's also a way to add you so that you can help set up some of the channels (not too sure what they're called here) then let me know as well and I can grant you access to that

I'll keep the link to it here for now, and I'll launch it as an announcement on the next release for everyone else who might be interested in joining

**Reply by Tealk** [reply]
What you linked there is a Space, a kind of folder for Rooms, though Tag might be a better fit because a Room can have multiple Spaces. So joining a Space is just having an overview of Rooms.

Now you would have to create rooms for the Space.


If you go to the Space on the gear, you can give people privileges under Roles, Admin and Moderator should be self-explanatory. 
![image](https://github.com/hjiangsu/thunder/assets/12276250/c1d6c5a4-5f8b-47b6-b7fc-1e21ecd25ab1)

The best is to create some rooms, often there is something like General and technic.
![image](https://github.com/hjiangsu/thunder/assets/12276250/14042551-933d-4d32-b86b-d58c6a5d2c58)

Rights you give in a space are not inherited by the rooms, so if you want to authorize me, you would have to do it individually for each room.

**Reply by hjiangsu** [reply]
Do you happen know know what permissions are given to admins vs moderators?

For now, I'll add you as a moderator for the space and the rooms to give you some ability to set stuff up as you see fit!

**Reply by Tealk** [reply]
> Do you happen know know what permissions are given to admins vs moderators?

What which rank can do you can see here:
![image](https://github.com/hjiangsu/thunder/assets/12276250/0500bb70-5684-4bff-9763-03397f844176)

**Reply by CTalvio** [reply]
Just found the matrix space, its a great way to organize some meta-discussions on the work going into the app.

Are there any rooms yet? It errored out after I joined and isn't loading any.

**Reply by hjiangsu** [reply]
There should be some rooms set up already! What's the error you're getting?

**Reply by CTalvio** [reply]
Simply "Failed to load list of rooms." But good to know its not supposed to be empty.

Probably down to something being up with my matrix instance. I'll say hi in matrix once I figure it out.

**Reply by Poolitzer** [reply]
looking forward to it

**Reply by Tealk** [reply]
@hjiangsu Are the rooms public?

To find them more easily you can select the following:
![image](https://github.com/hjiangsu/thunder/assets/12276250/f1789f61-6228-4f3e-a78f-2a3ed4fe4b7b)
